### ISSUE TITLE
cf push fails with "Staging time expired" if no buildpack is detected

### ISSUE THREAD

**Issue by kramerul** [cloudfoundry/cf-for-k8s#141]
Pushing the [dora app](https://github.com/cloudfoundry/cf-acceptance-tests/tree/master/assets/dora)

```
cf push dora -p workspace/cf-acceptance-tests/assets/dora
```

Fails with 

```
Error staging application: Staging time expired: staging failed
```

And the `kpack` `build` remains in step `detect`

```
status:
  conditions:
  - lastTransitionTime: "2020-04-14T13:30:44Z"
    status: "False"
    type: Succeeded
  observedGeneration: 1
  podName: e664f850-11d0-4191-bd77-edf2f05b26e1-build-1-xvk26-build-pod
  stack: {}
  stepStates:
  - terminated:
      containerID: docker://f314e52e4d260a346b99bf16c10effe81462c448c4344f01f4fdee95b82a36c3
      exitCode: 0
      finishedAt: "2020-04-14T13:30:41Z"
      reason: Completed
      startedAt: "2020-04-14T13:30:38Z"
  - terminated:
      containerID: docker://370e2a9fbc2f67350cb9845fd23e40051b2b83ab7dc42ce2ad6ce5656ea7bad1
      exitCode: 6
      finishedAt: "2020-04-14T13:30:44Z"
      reason: Error
      startedAt: "2020-04-14T13:30:41Z"
  - waiting:
      reason: PodInitializing
  - waiting:
      reason: PodInitializing
  - waiting:
      reason: PodInitializing
  - waiting:
      reason: PodInitializing
  stepsCompleted:
  - prepare
  - detect
```

It would be nice to get an error message that no matching buildpack was found.

**Comment by kramerul** [comment]
Seems to be related to https://cloudfoundry.slack.com/archives/CH9LF6V1P/p1586818920316400.

After enabling the `diego_docker` feature and pushing the dora app, The following error is shown:

```
Waiting for API to complete processing files...

Stopping app...

Staging app and tracing logs...
Error staging application: Stager error: Kpack build failed
FAILED
```

**Comment by Birdrock** [comment]
Hi @kramerul - thanks for reaching out!

The `diego_docker` feature flag was a known issue and has been resolved in master as of approximately a week ago (https://github.com/cloudfoundry/cf-for-k8s/issues/70).

The messaging regarding buildpack selection comes out of `kpack` via `pack`, so we cannot fix it directly. We can keep it as a known issue for the future. If you'd like some documentation surrounding debugging, we have started a document here that we plan to expand with additional common cases: https://github.com/cloudfoundry/cf-for-k8s/blob/438733fae9aaf15911181d00922a406f3ddbf238/docs/debugging.md
